include/wil/registry_basic.h

```diff
@@ -0,0 +1,1540 @@
+    }
+#endif
+
+        ::wil::unique_hkey create_unique_key(_In_ PCWSTR subKey, ::wil::registry::key_access access = ::wil::registry::key_access::read, _In_opt_ PCWSTR security_descriptor = nullptr) const
+        {
+            ::wil::unique_hkey local_key{};
+            create_key(subKey, local_key, access, security_descriptor);
+            return local_key;
+        }
+
+#if defined(__WIL_WINREG_STL)
+    ::wil::shared_hkey create_shared_key(_In_ PCWSTR subKey, ::wil::registry::key_access access = ::wil::registry::key_access::read, _In_opt_ PCWSTR security_descriptor = nullptr) const
+    {
+        ::wil::shared_hkey local_key{};
+        create_key(subKey, local_key, access, security_descriptor);
+        return local_key;
+    }
+#endif
+
+        typename err_policy::result delete_key(_In_opt_ PCWSTR sub_key) WI_NOEXCEPT
+        {
+            auto error = RegDeleteTreeW(::wil::registry::details::get_key(m_key), sub_key);
+            if (error == ERROR_FILE_NOT_FOUND)
+            {
+                error = ERROR_SUCCESS;
+            }
+            return err_policy::HResult(HRESULT_FROM_WIN32(error));
+        }
+
+        typename err_policy::result delete_value(HKEY key, _In_opt_ PCWSTR value_name) WI_NOEXCEPT
+        {
+            return err_policy::HResult(HRESULT_FROM_WIN32(RegDeleteValueW(key, value_name)));
+        }
+
+        template <typename R>
+        typename err_policy::result get_value(_In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, optional_value<R>& return_value, DWORD type = ::wil::registry::reg_view_details::get_value_type<R>()) const
+        {
+            for (;;)
+            {
+                DWORD data_size_bytes{::wil::registry::reg_view_details::get_buffer_size(return_value)};
+                auto error = ::RegGetValueW(
+                    ::wil::registry::details::get_key(m_key),
+                    subkey,
+                    value_name,
+                    ::wil::registry::details::get_value_flags_from_value_type(type),
+                    nullptr,
+                    ::wil::registry::reg_view_details::get_buffer(return_value),
+                    &data_size_bytes);
+                if (error == ERROR_SUCCESS)
+                {
+                    return_value.type = optional_value_status::has_value;
+                    break;
+                }
+                if (error == ERROR_FILE_NOT_FOUND)
+                {
+                    return_value.type = optional_value_status::no_value;
+                    break;
+                }
+                if (error == ERROR_MORE_DATA && ::wil::registry::reg_view_details::grow_buffer(return_value, data_size_bytes))
+                {
+                    // verify if grow_buffer succeeded allocation
+                    if (::wil::registry::reg_view_details::get_buffer_size(return_value) == 0)
+                    {
+                        error = ERROR_NOT_ENOUGH_MEMORY;
+                        break;

```

Comment:
Should this be an early return, not a break? It looks like this error is ignored.

Reply:
oops. this is a bug. thanks! fixing.